core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveRedundantJoin.java

```diff
@@ -57,15 +54,25 @@ public Result apply(JoinNode node, Captures captures, Context context)
 
     private boolean canRemoveJoin(Type joinType, boolean isLeftSourceEmpty, boolean isRightSourceEmpty)
     {
-        if (joinType == INNER && (isLeftSourceEmpty || isRightSourceEmpty)) {
-            return true;
-        }
-        if (joinType == LEFT && isLeftSourceEmpty) {
-            return true;
-        }
-        if (joinType == RIGHT && isRightSourceEmpty) {
-            return true;
+        switch (joinType) {
+            case INNER:
+                if (isLeftSourceEmpty || isRightSourceEmpty) {
+                    return true;
+                }
+            case LEFT:
```

Comment:
missing `break` or `return false` in all branches

Reply:
Do we need to have `break`. What if we could have simplify them like this 
```
case INNER:
    return isLeftSourceEmpty || isRightSourceEmpty
```
similarly for other branches too.